### PR TITLE
fix(ChangeStream): handle null changes

### PR DESCRIPTION
**Pull request by emadum**
## Description

[NODE-2626](https://jira.mongodb.org/browse/NODE-2626)

merge into master

**What changed?**

**Are there any files to ignore?**

### CODE REVIEW

**Comment by mbroadst** [comment]
@emadum is this still a draft?

**Comment by emadum** [comment]
> @emadum is this still a draft?

No just forgot to flip it after CI passed, ready for review!

File: src/cursor/core_cursor.ts [hunk 1]
```diff
@@ -473,50 +473,41 @@ class CoreCursor extends Readable {
...
+              cursor.ns = document.cursor.ns;
+            }
+            // Promote id to long if needed
+            cursor.cursorState.cursorId = typeof id === 'number' ? Long.fromNumber(id) : id;
+            cursor.cursorState.lastCursorId = cursor.cursorState.cursorId;
+            cursor.cursorState.operationTime = document.operationTime;
+
+            // If we have a firstBatch set it
+            if (Array.isArray(document.cursor.firstBatch)) {
+              cursor.cursorState.documents = document.cursor.firstBatch; //.reverse();
```
Comment (nbbeeken) [inline src/cursor/core_cursor.ts:505]: [nit] I know it was there before but we can cleanup the `// .reverse();` here right?

File: test/functional/change_stream.test.js [hunk 2]
```diff
@@ -2799,3 +2799,49 @@ describe('Change Stream Resume Error Tests', function() {
     })
   });
 });
+context('NODE-2626', function() {
```
Comment (nbbeeken) [inline test/functional/change_stream.test.js:2802]: Maybe add the name of the ticket just to make this a bit more descriptive

Comment (emadum) [reply test/functional/change_stream.test.js:2802]: Sounds good. The name of the ticket was not very descriptive of the actual issue, so I'll use the name of the PR.

**Pull request merged**